APIDesafioWarren/Controllers/CustomerController.cs

```diff
@@ -242,15 +244,9 @@ public IActionResult Delete(int id)
         {
             return SafeAction(() =>
             {
-                var cli = _customerServices
-                        .GetBy(c => c.Id == id);
-
-                if (cli is null)
-                {
-                    return NotFound("Customer not found!");
-                }
-                _customerServices.Remove(cli);
-                return Ok("Customer removed!");
+                return !_customerServices.Remove(id)
+                 ? BadRequest("Fail to remove a customer")
+                 : NoContent();
```

Comment:
Qual o motivo da falha?

```suggestion
                return !_customerServices.Remove(id)
                    ? BadRequest("Fail to remove a customer")
                    : NoContent();
```